src/Compress/AggregateArchiveAdapterResolver.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter\Compress;
+
+use Laminas\Filter\Exception\RuntimeException;
+use Laminas\Filter\File\FileInformation;
+
+use function array_values;
+use function sprintf;
+
+final class AggregateArchiveAdapterResolver implements ArchiveAdapterResolverInterface
```

Comment:
This "Aggregate Adapter Resolver" can be used to compose a chain of concrete adapter resolvers. The first successful resolver wins.